src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1148,6 +1149,7 @@ private static DocumentTombstone TableValueToTombstone(JsonOperationContext cont
                     }
                     table.Delete(doc.StorageId);
 
+                    // TODO: Do not delete the attachment streams if versioning is on
```

Comment:
DeleteAttachmentsOfDocument also delete the streams. But streams should not be deleted if there is a versioned attachment metadata with this stream. And I'll add another todo, not to delete if there is same hash in a different metadata.